verification/curator-service/api/test/clients/aws-events-client.test.ts

```diff
@@ -2,58 +2,114 @@ import AWS from 'aws-sdk';
+});
+const removeTargetsSpy = jest.fn().mockResolvedValue({
+    FailedEntries: [],
+    FailedEntryCount: 0,
+});
+
 beforeAll(() => {
     AWSMock.setSDKInstance(AWS);
 });
 
+beforeEach(() => {
+    deleteRuleSpy.mockClear();
+    putRuleSpy.mockClear();
+    putTargetsSpy.mockClear();
+    removeTargetsSpy.mockClear();
+    AWSMock.mock('CloudWatchEvents', 'deleteRule', deleteRuleSpy);
+    AWSMock.mock('CloudWatchEvents', 'putRule', putRuleSpy);
+    AWSMock.mock('CloudWatchEvents', 'putTargets', putTargetsSpy);
+    AWSMock.mock('CloudWatchEvents', 'removeTargets', removeTargetsSpy);
+});
+
 afterEach(() => {
     AWSMock.restore('CloudWatchEvents');
 });
 
 describe('putRule', () => {
-    it('returns created AWS CloudWatch Events rule ARN', async () => {
+    it('returns subject AWS CloudWatch Events rule ARN', async () => {
         const expectedArn = 'expectedArn';
-        AWSMock.mock(
-            'CloudWatchEvents',
-            'putRule',
-            (params: any, callback: Function) => {
-                callback(null, { RuleArn: expectedArn });
-            },
+        putRuleSpy.mockResolvedValueOnce({ RuleArn: expectedArn });
+        const client = new AwsEventsClient(
+            'us-east-1',
+            fakeRetrievalFunctionArn,
         );
-        const client = new AwsEventsClient('us-east-1');
 
         const ruleArn = await client.putRule(
             'passingRule',
             'description',
             'rate(1 hour)',
         );
         expect(ruleArn).toEqual(expectedArn);
+        expect(putRuleSpy).toHaveBeenCalledTimes(1);
+    });
+    it('creates a target for the rule if targetId is provided', async () => {
+        const client = new AwsEventsClient(
+            'us-east-1',
+            fakeRetrievalFunctionArn,
+        );
+
+        await client.putRule(
+            'passingRule',
+            'description',
+            'rate(1 hour)',
+            'targetId',
+        );
+        expect(putTargetsSpy).toHaveBeenCalledTimes(1);
     });
-    it('throws errors from AWS', async () => {
+    it('does not mutate rule targets if targetId not provided', async () => {
+        const client = new AwsEventsClient(
+            'us-east-1',
+            fakeRetrievalFunctionArn,
+        );
+
+        await client.putRule('passingRule', 'description', 'rate(1 hour)');
+        expect(putTargetsSpy).not.toHaveBeenCalled();
+    });
+    it('throws errors from AWS putRule call', async () => {
         const expectedError = new Error('AWS error');
-        AWSMock.mock(
-            'CloudWatchEvents',
-            'putRule',
-            (params: any, callback: Function) => {
-                callback(expectedError, null);
-            },
+        putRuleSpy.mockRejectedValueOnce(expectedError);
+        const client = new AwsEventsClient(
+            'us-east-1',
+            fakeRetrievalFunctionArn,
         );
-        const client = new AwsEventsClient('us-east-1');
 
         expect.assertions(1);
```

Comment:
not sure you need this with the toThrow?

Reply:
It's certainly less useful with `.rejects`, which automatically fails if the promise is fulfilled. I'd thought it would still protect us against forgetting to return or await the promise -- but I've just tested this, and it doesn't seem to be the case. I'll remove throughout.